content/ArtificialIntelligence.txt

```diff
@@ -503,21 +503,54 @@ Problems in training an RNN
 
  - What is a recurrent neural network (RNN)?
 
+What is an attention head?
+
+  An attention head (also described [here](https://towardsdatascience.com/all-you-need-to-know-about-attention-and-transformers-in-depth-understanding-part-1-552f0b41d021#8607)) takes as input a list of embeddings and produces as output a list of attention values.
+
+  An attention head has three matrices of parameters that can be learned like other parameters in a neural network. Let's call these matrices P1, P2, and P3.
+
+  When evaluating an attention head for a specific input, we first make three copies of the input. One copy is called "Queries", one copy is called "Keys", and one copy is called "Values".
+
+  Then, we multiply Queries by P1, Keys by P2, and Values by P3.
+
+  Then, for each token in Queries, we compute the similarity of that query to each other key via a dot product.
```

Comment:
I assume here the Queries = Original Queries * P1. Maybe we can say something like the new Queries?

Reply:
Made some changes in the next version of this commit